src/java.desktop/unix/native/libawt_xawt/awt/awt_GraphicsEnv.c

```diff
@@ -1425,6 +1425,9 @@ Java_sun_awt_X11GraphicsDevice_getDoubleBufferVisuals(JNIEnv *env,
     AWT_FLUSH_UNLOCK();
     for (i = 0; i < visScreenInfo->count; i++) {
         XdbeVisualInfo* visInfo = visScreenInfo->visinfo;
+        if ((*env)->ExceptionCheck(env)) {
+            break;
+        }
         (*env)->CallVoidMethod(env, this, midAddVisual, (visInfo[i]).visual);
         if ((*env)->ExceptionCheck(env)) {
```

Comment:
Looks like this `if` block can be removed, since it is effectively handled by check above.

Reply:
No .. that check is needed since CallVoidMethod() could raise an exception

Reply:
Sure, it can, but:

- if there is another iteration to do it will be handled by check on line `1428` and will break the cycle.
- if it is the last iteration there is no need to break since `CallVoidMethod()` is last statement in cycle.

Do I missing something?